hack/generator/pkg/codegen/pipeline/assert_types_collection_valid.go

```diff
@@ -3,21 +3,21 @@
  * Licensed under the MIT license.
  */
 
-package codegen
+package pipeline
 
 import (
 	"context"
 
 	"github.com/Azure/azure-service-operator/hack/generator/pkg/astmodel"
 )
 
-// assertTypesCollectionValid creates a PipelineStage that ensures that each reachable type in the types collection
+// AssertTypesCollectionValid creates a Stage that ensures that each reachable type in the types collection
 // has TypeName's that are all reachable as well. This check fails if there is any TypeName that refers to a type that doesn't
 // exist.
-func assertTypesCollectionValid() PipelineStage {
-	return MakePipelineStage(
+func AssertTypesCollectionValid() Stage {
+	return MakeStage(
 		"assertTypesStructureValid",
-		"Asserts that the types collection is valid",
+		"Verify that all named types are properly defined",
```

Comment:
```suggestion
		"Verify that all local TypeNames refer to a type",
```

What you have now makes it sound like it's looking at the structure of named types when really it's looking at the references between types (named or not)